app/lib/common/widgets/html_editor/components/mention_list.dart

```diff
@@ -0,0 +1,289 @@
+import 'package:acter/common/providers/room_providers.dart';
+import 'package:acter/common/utils/constants.dart';
+import 'package:acter/common/widgets/html_editor/components/mention_item.dart';
+import 'package:acter_avatar/acter_avatar.dart';
+import 'package:flutter_gen/gen_l10n/l10n.dart';
+import 'package:acter/common/widgets/html_editor/models/mention_block_keys.dart';
+import 'package:acter/common/widgets/html_editor/models/mention_type.dart';
+import 'package:acter/features/chat_ng/providers/chat_room_messages_provider.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:appflowy_editor/appflowy_editor.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+
+class MentionList extends ConsumerStatefulWidget {
+  const MentionList({
+    super.key,
+    required this.editorState,
+    required this.roomId,
+    required this.mentionType,
+    required this.onDismiss,
+  });
+
+  final EditorState editorState;
+  final String roomId;
+  final MentionType mentionType;
+  final VoidCallback onDismiss;
+
+  @override
+  ConsumerState<MentionList> createState() => _MentionHandlerState();
+}
+
+class _MentionHandlerState extends ConsumerState<MentionList> {
+  final _focusNode = FocusNode();
+  final _scrollController = ScrollController();
+
+  int _selectedIndex = 0;
+
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance.addPostFrameCallback((_) {
+      _focusNode.requestFocus();
+    });
+  }
+
+  @override
+  void dispose() {
+    widget.onDismiss();
+    _scrollController.dispose();
+    _focusNode.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final isDesktop = desktopPlatforms.contains(Theme.of(context).platform);
```

Comment:
reminder: this is almost always bad. checking for the specific feature you want is probably the better idea. In this case: check for/directly use the actual hardware keyboard..